parser: correctly parse LINESTRINGZ/M/ZM

otan opened the pull request:
Looks like everything else was done except LINESTRING!

Resolves #70653

Release note (sql change): Fixed a bug where LINESTRINGZ, LINESTRINGZM
and LINESTRINGM could not be used as a column type.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/70654)
<!-- Reviewable:end -->

otan committed:
parser: correctly parse LINESTRINGZ/M/ZM

Looks like everything else was done except LINESTRING!

Release note (sql change): Fixed a bug where LINESTRINGZ, LINESTRINGZM
and LINESTRINGM could not be used as a column type.

rafiss reviewed (approved):
😂

otan commented:
bors r=rafiss
